subxt/src/extrinsic/params.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+use codec::{
+    Compact,
+    Encode,
+};
+
+use crate::{
+    Config,
+    Encoded,
+};
+
+// We require Era as a param below, so make it available from here.
+pub use sp_runtime::generic::Era;
+
+/// This trait allows you to configure the "signed extra" and
+/// "additional" parameters that are signed and used in transactions.
+/// see [`BaseExtrinsicParams`] for an implementation that is compatible with
+/// a Polkadot node.
+pub trait ExtrinsicParams<T: Config> {
+    /// Thexe parameters can be provided to the constructor along with
+    /// some default parameters that `subxt` understands, in order to
+    /// help construct your [`ExtrinsicParams`] object.
+    type OtherParams;
+
+    /// Construct a new instance of our [`ExtrinsicParams`]
+    fn new(
+        spec_version: u32,
+        tx_version: u32,
+        nonce: T::Index,
+        genesis_hash: T::Hash,
+        other_params: Self::OtherParams,
+    ) -> Self;
+
+    /// This is expected to SCALE encode the "signed extra" parameters
+    /// to some buffer that has been provided. These are the parameters
+    /// which are sent along with the transaction, as well as taken into
+    /// account when signing the transaction.
```

Comment:
These are not taken into account when signing a transactions, the encoded extra data is part of the transaction payload. This topic is very confusing so it is important to be clear here.

Reply:
Perhaps I wasn't clear; what I meant here was that these are part of the payload of bytes that is signed (call, extra, additional) as well as being sent along with the transaction (whereas the additional params belowaren't sent but are equally taken into account, ie included i nthe signed payload)

Reply:
You are right, I misread.